cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala

```diff
@@ -442,12 +447,69 @@ class BreadthFirstTraverser[+V <: Node](val graph: Graph[V], val dir: GraphDir,
     super.visitNode(node)
   }
 
-  def shouldBeEnqueued(color: Walk.NodeColor.Color): Boolean = {
+  def shouldBeEnqueued(color: Color): Boolean = {
     color == Walk.NodeColor.Unenqueued
   }
 }
 
 /**
+ * Traverses in BFS order and keeps track of all shortest paths from any node in `homeNodeIds` to
+ * all target paths.  The traversal happens along all outbound nodes, it is suggested that `graph`
+ * storage direction is `GraphDir.OutDir`
+ * @param graph The graph to traverse on.  This graph need not necessarily be directed, but the
+ *              traversal will happen along all outbound nodes.
+ * @param homeNodeIds The sequence of source nodes to find all shortest paths from.  Each shortest path must
+ *                    start with one of the home nodes.
+ * @param dir The direction of the graph traversal
+ * @param walkLimits  How the graph traversal will be limited
+ * @tparam V A subtype of `Node`
+ */
+class ShortestPathsTraverser[+V <: Node](val graph: DirectedGraph[V], val homeNodeIds: Seq[Int],
+                                         val dir: GraphDir = GraphDir.OutDir,
+                                         walkLimits: Walk.Limits = Walk.Limits())
+  extends DepthTracker with BoundedIterator[V] {
+
+  // Queueing parameters
+  override def nodePriority = GraphTraverserNodePriority.FIFO
+  override lazy val limits = walkLimits
+
+  val ancestors = mutable.Map.empty[Int, Seq[Int]]
```

Comment:
Do you want a Seq[Int] or a Set[Int] here?

Have you looked at com.twitter.cassovary.graph.tourist.InfoKeeper which is intended to keep information per node?
e.g., you can reuse com.twitter.cassovary.graph.tourist.PrevNbrCounter which keeps an Int2IntMap per node. For your case, for each node n, the Int2IntMap will contain as key an ancestor of node n and value of 1.

alternatively, you can implement and use a new SetIntsInfoKeeper
class SetIntsInfoKeeper(override val onlyOnce: Boolean) extends InfoKeeper[Set[Int]] {
  protected val infoPerNode =  new mutable.Map.empty[Int, Set[Int]]

 ...

recordSetMember(nodeId: Int, setMemberId: Int) {
infoPerNode(nodeId).add(setMemberId)
}
}